db/migration/1657627850269-TargetYear.ts

```diff
@@ -0,0 +1,35 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class TargetYear1657627850269 implements MigrationInterface {
+    name = "TargetYear1657627850269"
+
+    async runAll(queryRunner: QueryRunner, query: string): Promise<void> {
+        // There is no way for a query to target all indices of a JSON array, so we just run it multiple times.
+        for (let i = 0; i < 50; i++) {
+            const currentQuery = query.replace(/§§/g, i.toString())
+            await queryRunner.query(currentQuery)
+        }
```

Comment:
Does this solve the problem? https://github.com/owid/owid-grapher/blob/31684fe579b1e32868d4968ff3cf7dc4bf40e58b/db/migration/1646223315110-DropDisplayForSizeDimension.ts#L7-L16

And if it does, would you rewrite the migration, or prefer not to spend time and run this instead?

Reply:
That is very clever indeed! However, I just tried it, and it doesn't work for this case:

If there is more than one dimension that needs to be corrected, we need an iterative procedure that iteratively updates the `n+1`-th dimension based on the updated `n`-th dimension.
However, your solution will only update one dimension in a run (because `config` is on the row-level, and updates to it don't propagate to other rows).

Reply:
Ah interesting, makes sense, thanks for checking!